City-of-Helsinki/drupal-helfi-kasvatus-koulutus
# UHF-5106: Full pagination

# Full pagination [UHF-5106](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5106)
Enable full pagination to daycare search and group news archive.

## What was done
* Enabled full pagination for daycare search and group news archive views.

## How to install
* Checkout this branch and use latest database from KASKO testing.
* Update the HDBT theme
    * `composer require drupal/hdbt:dev-UHF-5106_full_pagination`
* Run `make drush-cim && make drush-cr`
* `make shell`
* Inside the container run `drush locale:check; drush locale:update; drush cr`

## How to test
* Go to unit entity listing and search for Daycare. Select all that it lists on the first page and publish them all.
* Next select them again and add the "Daycare" term as the Unit type on each of them. After doing this you should have at least some daycare units. The stupid bulk operations bugs so you might have to do this a few times and always unselect the ones that it applied the changes.
* Go to edit the daycare search view https://helfi-kasko.docker.so/en/childhood-and-education/admin/structure/views/view/daycare_search and change the pagination settings so that you can check all kinds of combinations for the pager (set for example the number of items displayed to 3 to get a lot of pages etc.)
* Now go see the pager in the test page that is on latest testing db: https://helfi-kasko.docker.so/en/childhood-and-education/daycare-search-test-page
* You can translate the page as well to see the translations working on the pager.
* Go to edit the group news archive view https://helfi-kasko.docker.so/en/childhood-and-education/admin/structure/views/view/group_news_archive and change the pagination settings so that you can check all kinds of combinations for the pager (set for example the number of items displayed to 3 to get a lot of pages etc.)
* Go to group nodes of the Kuvataidelukio https://helfi-kasko.docker.so/en/childhood-and-education/group/1/nodes and from there check the Teron testi uutisarkisto page: https://helfi-kasko.docker.so/fi/kasvatus-ja-koulutus/helsingin-kuvataidelukio/teron-testi-uutisarkisto it should have the pager up and running as well.

## Other PRs
* https://github.com/City-of-Helsinki/drupal-hdbt/pull/303
